recipes/14/14_regular_joins.md

```diff
@@ -0,0 +1,47 @@
+# 14 Regular Joins
+
+:bulb: This example will show how you can use joins to correlate rows across multiple tables.
+
+Flink SQL supports complex and flexible join operations over continuous tables.
+There are several different types of joins to account for the wide variety of semantics queries may require.
+
+Regular joins are the most generic and flexible type of join.
+These include the standard `INNER` and `[FULL|LEFT|RIGHT] OUTER` joins that are available in most modern databases. 
+
+Suppose we have a [NOC list](https://en.wikipedia.org/wiki/Non-official_cover) of secret agents all over the world.
+Your mission if you choose to accept it, is to join this table with another containin the agents real name.
+
+In Flink SQL, this can be achieved using a simple `INNER JOIN`.
+Flink will join the tables using an equi-join predicate on the `agent_id` and output a new row everytime there is a match.
+
+```sql
+CREATE TABLE NOC (
+  agent_id STRING,
+  codename STRING
+)
+WITH (
+  'connector' = 'faker',
+  'fields.agent_id.expression' = '#{regexify ''(1|2|3|4|5){1}''}',
+  'fields.codename.expression' = '#{superhero.name}',
+  'rows-per-second' = '10'
+)
+
+CREATE TABLE RealNames (
+  agent_id STRING,
+  name     STRING
+)
+WITH (
+  'connector' = 'faker',
+  'fields.agent_id.expression' = '#{regexify ''(1|2|3|4|5){1}''}',
+  'fields.name.expression' = '#{harry_potter.characters}',
```

Comment:
You might be stretching it a bit with Harry Potter names as the RealNames :)

Reply:
I just needed some names 🤷